### PR TITLE
Simplified Shop Use *(may need testing w/ 2+ players)*

### PR DESCRIPTION
**Pull request by ThunderRazor**
PATCH NOTES:
- All shops are closed to players under 5000 xp.
- Designated shops are open to players with 5000 xp.
- Easy Company shops display xp requirement at the beginning when players under 5000 xp are present in non-recruit modes.
- Removed xp requirement tooltip for all shop items.
- Corrected formatting consistencies for all item tooltips.

DETAILS:
- Added validator to "Shop Share" ability to check player XP (custom resource).
- Added function "EC Shop XP Req" which checks if any players under 5000 xp are present; adds Unit Info Text "Shop Requirement: 5000 XP" to the shops in Easy Company (non-recruit modes).
- Removed xp requirement tooltips for all shop items (since 5000 xp requirement is universal).

ADDITIONAL NOTES:
- The *Shop Open / Shop Closed* to individual players depending on xp may need to be tested with 2 or more players.
- *In theory*, the shop should be closed to players under 5000 xp while open to players with 5000 xp *simultaneously.*
- Designated shops in certain story modes and difficulty modes remain the same.

### CODE REVIEW

File: src/NOTD.SC2Map/Base.SC2Data/GameData/AbilData.xml [hunk 1]
```diff
@@ -3154,6 +3154,7 @@
         <Range value="5"/>
         <AutoCastRange value="5"/>
         <TargetFilters value="Heroic;Neutral,Enemy"/>
+        <AutoCastValidatorArray value="PlayerHas5000XP"/>
```
Comment (ThunderRazor) [inline src/NOTD.SC2Map/Base.SC2Data/GameData/AbilData.xml:3157]: The shop will be open (shared control) to the target player (shop user) if the validator is true.

File: src/NOTD.SC2Map/Base.SC2Data/GameData/ValidatorData.xml [hunk 2]
```diff
@@ -551,6 +551,12 @@
         <CombineArray value="UnitisPathfinder"/>
         <CombineArray value="TargetIsEnemyorNeutral"/>
     </CValidatorCombine>
+    <CValidatorPlayerCompareResource id="PlayerHas5000XP">
+        <WhichPlayer Value="Target"/>
+        <Compare value="GE"/>
+        <Resource value="Custom"/>
+        <Value value="5000"/>
+    </CValidatorPlayerCompareResource>
```
Comment (ThunderRazor) [inline src/NOTD.SC2Map/Base.SC2Data/GameData/ValidatorData.xml:559]: This is the validator that checks player XP.
"Custom" resource *is* NOTD "XP", "GE" is ">=" (greater than or equal to)

File: src/NOTD.SC2Map/MapScript.galaxy [hunk 3]
```diff
@@ -6660,6 +6661,29 @@ bool auto_gf_ECEnvironmentalChanges_TriggerFunc (bool testConds, bool runActions
...
+    for ( ; !PlayerGroupLoopDone() ; PlayerGroupLoopStep() ) {
+        if ((PlayerGetPropertyInt(PlayerGroupLoopCurrent(), c_playerPropCustom) < 5000)) {
+            lv_vets = false;
+        }
+
+    }
+    PlayerGroupLoopEnd();
+    if ((lv_vets == false)) {
+        UnitSetInfoText2(UnitFromId(178), StringExternal("Param/Value/F64869B1"));
+        UnitSetInfoText2(UnitFromId(2503), StringExternal("Param/Value/97BA6CAF"));
```
Comment (ThunderRazor) [inline src/NOTD.SC2Map/MapScript.galaxy:6682]: Check all players to see if anyone is under 5000 xp. If any player is under 5000 xp (in Easy Company non-recruit modes), the XP requirement will be displayed to the Troop Transport and the Pipedream's Emporium. This is intended for "first exposure" to vet games for low xp players, which typically takes place in Easy Company.

File: src/NOTD.SC2Map/MapScript.galaxy [hunk 4]
```diff
@@ -12343,9 +12367,9 @@ void gf_RestoreLeaverControl () {
     }
 }
 
-void gf_SetCombatKnifeDamage (int lp_player) {
+void gf_SetCombatKnifeDamage (int lp_triggeringPlayer) {
     // Implementation
-    CatalogFieldValueModify(c_gameCatalogEffect, "CombatKnifeDamage", "Amount", lp_player, FixedToString(((IntToFixed(gv_playerRank[lp_player]) - 1.0) * 0.5), c_fixedPrecisionAny), c_upgradeOperationAdd);
+    CatalogFieldValueModify(c_gameCatalogEffect, "CombatKnifeDamage", "Amount", lp_triggeringPlayer, FixedToString(((IntToFixed(gv_playerRank[lp_triggeringPlayer]) - 1.0) * 0.5), c_fixedPrecisionAny), c_upgradeOperationAdd);
 }
```
Comment (ThunderRazor) [inline src/NOTD.SC2Map/MapScript.galaxy:12373]: I have no idea what's going on here. I never touched this function, and Galaxy Editor decided to change the parameter name (which does not affect the mechanics at all).

Comment (ArcanePariah) [reply src/NOTD.SC2Map/MapScript.galaxy:12373]: I changed it, so it probably got updated when you pulled in the current main branch

**Comment by dave-sWpR46Ae** [comment]
I would prefer to leave the option to purchase kittens at 500 xp.  Alternatively, we can make all shop purchases 500 xp.

**Comment by ThunderRazor** [comment]
Keep in mind that I have not changed the XP requirements of any purchases. I simply made the shops unusable for all players who do not meet the [5000] xp requirement.
Since kittens are purchased at a different shop than the usual items it could be given a different xp requirement, which would be shared with the BO insignia and mod points.

I really don't care what xp requirements should be utilized. I don't mind doing more work on the shops and shop functions. The important thing to me is that I clean it up, i.e. make the requirements and use consistent with the tooltips or the tooltips consistent with the requirements and use. That being said, regardless of what items have what requirements, I'll be more than happy to make it happen and to make it clean. @ArcanePariah just give me the word, and I'll make it happen.

**Comment by ReduxGelum** [comment]
Get rid of "ECShopXPReq". It's completely useless, just display the 5k xp message to everyone.

**Pull request merged**